### PR TITLE
Add example to plugin template

### PR DESCRIPTION
**Pull request by neonichu**
This allows running `bundle exec fastlane test` directly from the generated plugin directory and should encourage people to provide an example `Fastfile`.

### CODE REVIEW

File: fastlane/lib/fastlane/plugins/template/fastlane/Fastfile.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+lane :test do
+  <%= plugin_name %>()
+end
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/plugins/template/fastlane/Fastfile.erb]: Missing new line 💃

File: fastlane/lib/fastlane/plugins/template/fastlane/Pluginfile.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Autogenerated by fastlane
+
+gem '<%= gem_name %>', path: '.'
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/plugins/template/fastlane/Pluginfile.erb]: Missing new line 💃

**Comment by neonichu** [comment]
newlines ﻿ノ( ゜-゜ノ)

File: fastlane/lib/fastlane/plugins/template/README.md.erb [hunk 3]
```diff
@@ -14,6 +14,10 @@ fastlane add_plugin <%= plugin_name %>
 
 <%= summary %>
 
+## Example
+
+Check out the [example `Fastfile`](fastlane/Fastfile) to see how to use this plugin. You can also run it directly from a checkout of this repository using `bundle exec fastlane test`.
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/plugins/template/README.md.erb]: That requires the author of the plugin to set up a sample project, most of the times that's a full iOS/Android project. Maybe we should add more information here? I'd propose:

> Check out the [example `Fastfile`](fastlane/Fastfile) to see how to use this plugin. Try it by cloning the repo, running `fastlane install_plugins` and `bundle exec fastlane test`. 
> **Note to author:** Please set up a sample project to make it easy for users to explore what your plugin does. Provide everything that is necessary to try out the plugin in this project (including a sample Xcode/Android project if necessary)
 
What do you think @neonichu @mfurtak?

File: fastlane/lib/fastlane/plugins/template/.gitignore [hunk 4]
```diff
@@ -6,3 +6,5 @@ Gemfile.lock
 /_yardoc/
 /doc/
 /rdoc/
+fastlane/README.md
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/plugins/template/.gitignore]: Actually why not also commit the README.md? Wouldn't it make sense?

File: fastlane/lib/fastlane/plugins/template/README.md.erb (hunk 3 above)
Comment (neonichu) [inline fastlane/lib/fastlane/plugins/template/README.md.erb]: Makes sense

File: fastlane/lib/fastlane/plugins/template/.gitignore (hunk 4 above)
Comment (neonichu) [inline fastlane/lib/fastlane/plugins/template/.gitignore]: Feels redundant to the test in the gem's `README` — but we can also commit it, I have no preference.

**Pull request merged**